app/src/main/kotlin/co/netguru/baby/monitor/client/common/view/PulsatingView.kt

```diff
@@ -91,16 +91,34 @@ class PulsatingView(context: Context, attrs: AttributeSet?) : View(context, attr
 
     fun stop() {
         circlePaint.color = inactiveColor
-        animator?.end()
-        animator2?.end()
+        stopAnimators(listOf(firstPulseAnimator, secondPulseAnimator))
         invalidate()
     }
 
+    private fun stopAnimators(animators: List<ValueAnimator?>) {
+        animators.forEach {
+            it?.run {
+                end()
+                removeAllUpdateListeners()
+            }
+        }
+    }
+
+    override fun onDetachedFromWindow() {
+        super.onDetachedFromWindow()
+        stop()
+    }
+
     override fun onDraw(canvas: Canvas?) {
         canvas ?: return
         canvas.drawCircle(width / 2f, height / 2f, circleSize / 2f, circlePaint)
-        canvas.drawCircle(width / 2f, height / 2f, (height / 2f) * size, paint)
-        canvas.drawCircle(width / 2f, height / 2f, (height / 2f) * size2, paint2)
+        canvas.drawCircle(width / 2f, height / 2f, (height / 2f) * firstPulseSize, firstPulsePaint)
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/netguru/baby-monitor-client-android/blob/85cb1800bcef802a6706010d0e78f2363392f785/app/src/main/kotlin/co/netguru/baby/monitor/client/common/view/PulsatingView.kt#L115"></span>Parentheses in (height / 2f) are unnecessary and can be replaced with: height / 2f</td>
    </tr>
  </tbody>
</table>